# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/gcc/hotfix-glibc-2.42.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

../../../../libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp:489:31: error: invalid application of 'sizeof' to incomplete type '__sanitizer::termio'
  489 |   unsigned struct_termio_sz = sizeof(struct termio);
      |                               ^~~~~
#if SANITIZER_GLIBC || SANITIZER_ANDROID
  unsigned struct_termio_sz = sizeof(struct termio);
#endif

../../../../libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp:768:27: error: invalid application of 'sizeof' to incomplete type '__sanitizer::termio'
../../../../libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp:772:27: error: invalid application of 'sizeof' to incomplete type '__sanitizer::termio'
../../../../libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp:773:28: error: invalid application of 'sizeof' to incomplete type '__sanitizer::termio'
../../../../libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp:774:28: error: invalid application of 'sizeof' to incomplete type '__sanitizer::termio'

--- gcc-15.1.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp.vanilla	2025-07-29 13:09:27.591641995 +0200
+++ gcc-15.1.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp	2025-07-29 13:12:41.136946537 +0200
@@ -486,7 +486,7 @@
   unsigned struct_mtpos_sz = sizeof(struct mtpos);
   unsigned struct_rtentry_sz = sizeof(struct rtentry);
 #if SANITIZER_GLIBC || SANITIZER_ANDROID
-  unsigned struct_termio_sz = sizeof(struct termio);
+  //unsigned struct_termio_sz = sizeof(struct termio);
 #endif
   unsigned struct_vt_consize_sz = sizeof(struct vt_consize);
   unsigned struct_vt_sizes_sz = sizeof(struct vt_sizes);
--- gcc-15.1.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp.vanilla	2025-08-01 11:58:09.509584223 +0200
+++ gcc-15.1.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp	2025-08-01 12:04:04.336766966 +0200
@@ -765,13 +765,13 @@
   unsigned IOCTL_SOUND_PCM_WRITE_FILTER = SOUND_PCM_WRITE_FILTER;
 #endif // SOUND_VERSION
   unsigned IOCTL_TCFLSH = TCFLSH;
-  unsigned IOCTL_TCGETA = TCGETA;
+  //unsigned IOCTL_TCGETA = TCGETA;
   unsigned IOCTL_TCGETS = TCGETS;
   unsigned IOCTL_TCSBRK = TCSBRK;
   unsigned IOCTL_TCSBRKP = TCSBRKP;
-  unsigned IOCTL_TCSETA = TCSETA;
-  unsigned IOCTL_TCSETAF = TCSETAF;
-  unsigned IOCTL_TCSETAW = TCSETAW;
+  //unsigned IOCTL_TCSETA = TCSETA;
+  //unsigned IOCTL_TCSETAF = TCSETAF;
+  //unsigned IOCTL_TCSETAW = TCSETAW;
   unsigned IOCTL_TCSETS = TCSETS;
   unsigned IOCTL_TCSETSF = TCSETSF;
   unsigned IOCTL_TCSETSW = TCSETSW;
